modules/core/04-channel/types/upgrade.go

```diff
@@ -69,3 +71,22 @@ func (uf UpgradeFields) ValidateBasic() error {
 func (ut UpgradeTimeout) IsValid() bool {
 	return !ut.Height.IsZero() || ut.Timestamp != 0
 }
+
+// HasPassed returns true if the upgrade has passed the timeout height or timestamp
+func (ut UpgradeTimeout) HasPassed(ctx sdk.Context) (bool, error) {
```

Comment:
I'm not sure how I feel about this API, it feels arbitrary to me to return an error when checking if the timeout has passed. I would prefer to ensure we never call `HasPassed` on an invalid timeout and then just return supplementary information when checking if the timeout has passed. It isn't an error if the timeout has passed or not. Was there a reason we didn't use:
```go
func (ut UpgradeTimeout) HasPassed(ctx sdk.Context) (bool, string) 
```
where string is information about the timeout relative to the current time. Then:
```go
// recv packet
if hasPassed, info := timeout.HasPassed(ctx); hasPassed {
    return errorsmod.Wrapf(ErrTimeoutPassed, info)
}
``` 
or 
```go
// timeout packet
if hasPassed, info := timeout.HasPassed(ctx); !hasPassed {
    return errorsmod.Wrapf(ErrInvalidTimeout, "timeout has not passed: %s", info)
}

Reply:
I was also going to raise a comment about this API. To be honest, I initially expected this function to just return a `bool`, and was unsure of the value of the "info" string. 

But I see now it can add valuable error info indicating whether its the block height or timestamp which has elapsed. 

I'd be happy with `bool` or `(bool, string)` but I do agree that `(bool, error)` seems a little off

Reply:
I am happy with `bool, info` as I do feel like it is important to indicate which timeout has elapsed.